core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -32,6 +50,7 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
 
     sealed class ResolveResult {
```

Comment:
I'd simplify this by assuming that the result is "success" iff there are no errors:
```
data class ResolutionResult(
    val dependencies: ScriptDependencies, // ScriptDependencies.Empty in case of failure
    val diagnostics: List<ScriptDiagnostic>
) {
    val isSuccess: Boolean get() = diagnostics.none { it.severity == ERROR } // maybe an extension
}
```